TCP probe health check for Device State Manager (#121)
observability-lab-cse/observability-lab: sample-application/devices-state-manager/DevicesStateManager/README.md

## Check results

> If you run the whole solution, including the Devices Simulator, then when Devices State Manager starts running, you should see messages being received and processed

Reviewer: ```suggestion
If you run the whole solution, including the Devices Simulator, then, once Devices State Manager starts running, you should see messages being received and processed:
```

```text
info: DevicesStateManager.EventHubReceiverService[0]
Received event: {"deviceId": "device-42", "deviceTimestamp": "2023-07-17T17:41:52.8690130Z", "temp": 23.70572735914296}
```

## Health check

> Devices State Manager has health-check implemented as a tcp probe so that kubernetes can use it in the readiness probe. If you want to check health-check locally you can execute following steps:

Reviewer: ```suggestion
Devices State Manager has health-check implemented as a TCP probe so that Kubernetes can use it in the readiness probe. If you want to check the health-check locally, you can execute following steps:
```

1. Run the Devices State Manager locally

> 2. In the application logs you should see a log indicating that the tcp probe service started:

Reviewer: ```suggestion
2. In the application logs you should see a message indicating that the TCP probe service started:
```

```
info: DevicesStateManager.TcpHealthProbeService[0]
Started health check service.
```

> 3. Request tcp connection on port 8090, you can use e.g. nc command from terminal:

Reviewer: ```suggestion
3. Request a TCP connection on port 8090, you can use e.g. `nc` command from terminal:
```

```bash
nc localhost 8090
```

> 4. Now in your application logs you should see a message:

Reviewer: ```suggestion
4. Now, in your application logs you should see the following message:
```
